io.c

```diff
@@ -2457,15 +2457,16 @@ rb_io_eof(VALUE io)
 
 /*
  *  call-seq:
- *     ios.sync    -> true or false
+ *    sync -> true or false
  *
- *  Returns the current ``sync mode'' of <em>ios</em>. When sync mode is
- *  true, all output is immediately flushed to the underlying operating
- *  system and is not buffered by Ruby internally. See also
- *  IO#fsync.
+ *  Returns the value of the most recent call to IO#sync=,
+ *  or +false+ if the method has not been called; see IO#sync=:
```

Comment:
The previous language was more accurate. There are cases where the sync mode is set without `IO#sync=` (such as on the write pipe returned by `IO.pipe`).  I would  continue to state that this returns the sync mode, even if you only define the sync mode in `IO#sync=`

Reply:
Done.